rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -1112,8 +1127,8 @@ rmw_wait(
       RMW_SET_ERROR_MSG("read condition handle is null");
       return RMW_RET_ERROR;
     }
-    if (waitset.attach_condition(read_condition) != DDS::RETCODE_OK) {
-      RMW_SET_ERROR_MSG("failed to attach condition to waitset");
+    DDS::ReturnCode_t retcode = waitset.attach_condition(read_condition);
+    if (check_attach_condition_error(retcode) == RMW_RET_ERROR) {
```

Comment:
We should check for `!= RMW_RET_OK` instead since there are other error values.

Same below.